File: Iv Layhort/Scala-Layhort/src/Task_8/Exercise_1_AngryProfessor.scala
package Task_8
import io.StdIn._

object Exercise_1_AngryProfessor {
  def main(args: Array[String]): Unit = {
    println("Enter the number of test case: ")
    val testCase = readInt()
    for(_ <- 1 to testCase){
      println("Enter the number of students and cancellation threshold respectively")
      val convert = readLine().split(" ").map(_.toInt)
      val (numberOfStudents, cancellationThreshold) = (convert(0),convert(1))
      println("Enter the student arrival time respectively")
      var arrivalTime = readLine().split(" ")
      while(arrivalTime.length < numberOfStudents)
        arrivalTime = readLine().split(" ")
      println(if(arrivalTime.map(_.toInt).count(_ <= 0) <= cancellationThreshold) "YES" else "NO")
    }
  }
}

File: Iv Layhort/Java_Layhort/src/Task_8/Exercise_1_AngryProfessor.java
package Task_8;
import java.util.*;
public class Exercise_1_AngryProfessor {
    public static void main(String[] args) {
        Scanner input= new Scanner(System.in);
        int testCase,numberOfStudents,cancellationThreshold;
        System.out.println("Enter the number of test case:");
        testCase = input.nextInt();

        for(int i=0;i<testCase;i++){
            int onTime =0;
            System.out.println("Enter the number of students and cancellation threshold respectively :");
            numberOfStudents = input.nextInt();
            cancellationThreshold =input.nextInt();

            System.out.println("Enter the arrival time of students");
            for(int j=0;j<numberOfStudents;j++){
                int arrivalTime = input.nextInt();
                if(arrivalTime<=0) onTime++;
            }
            if(onTime<=numberOfStudents) System.out.println("YES"); else System.out.println("NO");
        }
    }
}
